### PR TITLE
Plg 27/bigger product images

### PR DESCRIPTION
**Pull request by nedick007**
Implemented new image helper in order to access product image base path (first image from product media gallery) and get a timestamp of last modification attached to the url.

### CODE REVIEW

File: Helper/ImagePathResolver.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Metrilo\Analytics\Helper;
+
+class ImagePathResolver extends \Magento\Framework\App\Helper\AbstractHelper
+{
+    /**
+     * Takes the best quality image url with timestamp of last modification
+     * @param $product / store product or quoteItem
```
Comment (adamof) [inline Helper/ImagePathResolver.php]: This won't work if you pass a quoteItem here, so this param is **always** product

File: Helper/ImagePathResolver.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     * @return string
+     */
+    public function getBaseImage($product) {
+
```
Comment (adamof) [inline Helper/ImagePathResolver.php]: remove empty line

File: Helper/ImagePathResolver.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            $imageMtime = strtotime(date("F d Y H:i:s.", filemtime($imagePath)));
+        }  
+
+        if(isset($imageUrl) && !empty($imageMtime)) {
+            $timestampUrl = $imageUrl . '?t=' . $imageMtime;
+        }
+
+        return $timestampUrl;
+    }
+
```
Comment (adamof) [inline Helper/ImagePathResolver.php]: remove empty line

File: Helper/OrderSerializer.php [hunk 4]
```diff
@@ -103,10 +103,9 @@ private function getProductDetails($quoteItem)
             }
 
             if ($product) {
-                if($product->getImage()) {
-                    $imageUrl = $this->imageHelperFactory->create()
-                        ->init($product, 'product_thumbnail_image')->getUrl();
-                    $dataItem['image_url'] = $imageUrl;
+
+                if(!empty($this->imagePathResolver->getBaseImage($product))) {
```
Comment (adamof) [inline Helper/OrderSerializer.php]: wouldn't this execute the `getBaseImage` code multiple times?

File: Model/Analytics.php [hunk 5]
```diff
@@ -91,11 +91,11 @@ public function addPageEvents()
                 'url'   => $product->getProductUrl()
             ];
             // Additional information ( image and categories )
-            if($product->getImage()) {
 
-                $imageUrl = $this->imageHelperFactory->create()
-                    ->init($product, 'product_thumbnail_image')->getUrl();
-                $data['image_url'] = $imageUrl;
+            if($product) {
+                if(!empty($this->imagePathResolver->getBaseImage($product))) {
```
Comment (adamof) [inline Model/Analytics.php]: same here, multiple times

**Comment by nedick007** [comment]
Code refactoring commit is ready for review.

**Pull request merged**